ext/src/zpages/tracez_data_aggregator.cc

```diff
@@ -0,0 +1,207 @@
+#include "opentelemetry/ext/zpages/tracez_data_aggregator.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext {
+namespace zpages {
+
+TracezDataAggregator::TracezDataAggregator(
+    std::shared_ptr<TracezSpanProcessor> span_processor,
+    long update_interval_in_milliseconds) {
+  tracez_span_processor_ = span_processor;
+
+  // Start a thread that calls AggregateSpans periodically or till notified.
+  execute_.store(true, std::memory_order_release);
+  aggregate_spans_thread_ =
+      std::thread([this, update_interval_in_milliseconds]() {
+        while (execute_.load(std::memory_order_acquire)) {
+          std::unique_lock<std::mutex> lock(mtx_);
+          AggregateSpans();
+          cv_.wait_for(lock, milliseconds(update_interval_in_milliseconds));
+        }
+      });
+}
+
+TracezDataAggregator::~TracezDataAggregator() {
+  // Notify and join the thread so object can be destroyed
+  if (execute_.load(std::memory_order_acquire)) {
+    execute_.store(false, std::memory_order_release);
+    cv_.notify_one();
+    aggregate_spans_thread_.join();
+  };
+}
+
+std::map<std::string, std::unique_ptr<TracezSpanData>>
+TracezDataAggregator::GetAggregatedTracezData() {
+  std::unique_lock<std::mutex> lock(mtx_);
+
+  /** NOTE: At the moment a copy of the aggregated data is returned from this
+   * getter to avoid concurrency issues. When it becomes more clear what type of
+   * object is needed by the zPage HTTP server (most likely a JSON),
+   * this function should be changed accordingily.
+   **/
+  std::map<std::string, std::unique_ptr<TracezSpanData>>
+      aggregated_tracez_data_cpy;
+  for (auto &name_to_data : aggregated_tracez_data_) {
+    std::string span_name = name_to_data.first;
+    aggregated_tracez_data_cpy[span_name] =
```

Comment:
I think aggregated_tracez_data_cpy can be std::map<std::string, TracezSpanData> instead. 
Do we need unique_ptr in this case as well?

Reply:
I don't think we need a unique ptr for that struct at all, as per Janet's earlier comment I am working on changing this